### PR TITLE
drop python2 support

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
@susebot run deploy

**Comment by tchaikov** [comment]
@susebot run deploy

**Comment by tchaikov** [comment]
@susebot run deploy

File: MANIFEST.in [hunk 1]
```diff
@@ -1,3 +1,2 @@
 include *.rst
-include requirements2.txt
```
Comment (kshtsk) [inline MANIFEST.in:2]: Since we drop py2 support, the requirements3.txt should be renamed back to requirements.txt

File: setup.py [hunk 2]
```diff
@@ -42,7 +42,7 @@
         'License :: OSI Approved :: MIT License',
         'Natural Language :: English',
         'Operating System :: POSIX :: Linux',
-        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3.6',
```
Comment (kshtsk) [inline setup.py:45]: Don't we support 3.5? Just to make it clear.

File: teuthology/misc.py [hunk 3]
```diff
@@ -647,7 +646,7 @@ def remove_lines_from_file(remote, path, line_is_valid_test,
     on when the main site goes up and down.
     """
     # read in the specified file
-    in_data = ensure_str(get_file(remote, path, False))
+    in_data = get_file(remote, path, False).decode()
```
Comment (kshtsk) [inline teuthology/misc.py:649]: I'm thinking about adding another method `Remote.read_file()` for usability.

File: setup.py (hunk 2 above)
Comment (tchaikov) [reply setup.py:45]: no, 3.5 is missing quite a few features. among other things the f-string. and some methods for testing, i cannot recall the exact names.

but 3.6 is a much more capable python3 than 3.5.

File: teuthology/orchestra/remote.py [hunk 4]
```diff
@@ -247,9 +245,9 @@ def sh(self, script, **kwargs):
             kwargs['stdout'] = BytesIO()
         if 'args' not in kwargs:
             kwargs['args'] = script
-        proc=self.run(**kwargs)
-        out=proc.stdout.getvalue()
-        return ensure_str(out)
+        proc = self.run(**kwargs)
+        out = proc.stdout.getvalue()
+        return out.decode()
```
Comment (kshtsk) [inline teuthology/orchestra/remote.py:250]: There should be a check if the stdout is BytesIO or StringIO to know if we need to apply or no `decode`.

File: MANIFEST.in (hunk 1 above)
Comment (tchaikov) [reply MANIFEST.in:2]: will do this in another iteration.

File: teuthology/orchestra/remote.py (hunk 4 above)
Comment (tchaikov) [reply teuthology/orchestra/remote.py:250]: fixed.

File: MANIFEST.in (hunk 1 above)
Comment (tchaikov) [reply MANIFEST.in:2]: done.

File: setup.py (hunk 2 above)
Comment (kshtsk) [reply setup.py:45]: thanks

File: tox.ini [hunk 5]
```diff
@@ -1,26 +1,12 @@
 [tox]
 envlist = docs, py3, py27, py27-integration, flake8, flake8-py3, openstack
```
Comment (kshtsk) [inline tox.ini:3]: The envlist from the `[tox]` section should be cleaned from the py27 targets.

Comment (kshtsk) [reply tox.ini:3]: btw, flake description should be replaced with flake8-py3 I guess.

Comment (tchaikov) [reply tox.ini:3]: fixed.

**Comment by tchaikov** [comment]
@susebot run deploy

File: tox.ini (hunk 5 above)
Comment (kshtsk) [reply tox.ini:3]: nice, what about flake8?

Comment (tchaikov) [reply tox.ini:3]: ahh, removed!

File: teuthology/misc.py (hunk 3 above)
Comment (tchaikov) [reply teuthology/misc.py:648]: BTW, read_file is a better name, IMHO.

**Comment by tchaikov** [comment]
@kshtsk mind taking another look?

**Comment by kshtsk** [comment]
Looks great, the tests passed too.
About to ask suse bot to another run.

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
https://ceph-ci.suse.de/job/teuthology-deploy-ovh/255/testReport/junit/(root)/teuthology/smoke_basic__clusters__fixed_3_cephfs_openstack__objectstore_bluestore_bitmap_supported_random_distro$__centos_8__tasks_rados_python_/attachments/teuthology-smoke-3.log 
Interesting failure.

**Comment by kshtsk** [comment]
@tchaikov it's green now,
thanks for the PR.

**Pull request merged**